### PR TITLE
Add NVCC path and Python version to `show_config`

### LINKED ISSUE #5210: Add NVCC path and Python version to show_config

### PR DESCRIPTION
**Pull request by asi1024**
Resolve #5210.

### CODE REVIEW

File: cupyx/_runtime.py [hunk 1]
```diff
@@ -215,6 +220,7 @@ def __str__(self):
             ('CuPy Version', self.cupy_version),
             ('NumPy Version', self.numpy_version),
             ('SciPy Version', self.scipy_version),
+            ('Python Version', platform.python_version()),
```
Comment (kmaehashi) [inline cupyx/_runtime.py:223]: Could you move this line to just after `OS`?

File: cupyx/_runtime.py [hunk 2]
```diff
@@ -215,6 +220,7 @@ def __str__(self):
             ('CuPy Version', self.cupy_version),
```
Comment (kmaehashi) [inline cupyx/_runtime.py:220]: Let's add platform information too.

```suggestion
            ('CuPy Version', self.cupy_version),
            ('CuPy Platform', 'NVIDIA CUDA' if not is_hip else 'AMD ROCm'),
```

File: cupyx/_runtime.py [hunk 3]
```diff
@@ -225,6 +231,11 @@ def __str__(self):
             ('CUDA Runtime Version', self.cuda_runtime_version),
         ]
 
+        compiler_name = 'hipcc' if is_hip else 'NVCC'
```
Comment (kmaehashi) [inline cupyx/_runtime.py:234]:
```suggestion
        compiler_name = 'hipcc' if is_hip else 'nvcc'
```

File: cupyx/_runtime.py [hunk 4]
```diff
@@ -225,6 +231,11 @@ def __str__(self):
...
+        records += [
+            (f'{compiler_name} PATH', self.nvcc_path)
```
Comment (kmaehashi) [inline cupyx/_runtime.py:236]: Could you move this line just after `CUDA Root`?

**Review by kmaehashi — CHANGES_REQUESTED** [review]

**Comment by asi1024** [comment]
@kmaehashi PTAL!

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Review by kmaehashi — APPROVED** [review]
LGTM!

**Pull request merged**